modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectRelationship/ObjectRelationshipFormBase.tsx

```diff
@@ -250,21 +250,7 @@ export function ObjectRelationshipFormBase({
 			setObjectDefinitions(objectDefinitions);
 		};
 
-		if (readonly) {
-			setObjectDefinitions([
-				{
-					externalReferenceCode:
-						values.objectDefinitionExternalReferenceCode2 as string,
```

Comment:
Hi @rebsilva! 🌷 just bringing some additional context here from the ticket and the fix, the values prop here points to relationship data and this is why we see the relationship label and not the definition one. setObjectDefinitions should point to all object definitions, which is done correctly through the fetch, without interfering with readonly expected behavior.
I've tested this manually with different and scenarios and ran objetcs suite on my personal repo to check if it wouldn't mess with any behavior I'm unaware of and it didn't. The condition that was causing the bug is 3 years old, so I suspect that probably this code made sense at the time but it doesn't anymore.
Look forward to your review, let me know if you have any doubts or finds any issue in this, thanks!! 💟

Reply:
thank you for the additional info lari! :purple_heart: